README.md

```diff
@@ -19,25 +19,50 @@ From your shell or command line:
+1. If your account gives you access to more than one tenant, select your account in the top right corner, and set your portal session to the desired Azure AD tenant
+   (using **Switch Directory**).
+1. In the left-hand navigation pane, select the **Azure Active Directory** service, and then select **App registrations (Preview)**.
 
+#### Register the client app (ConsoleApp-DeltaQuery-DotNet)
 
-###  Step 3: Setup sample project
+1. In **App registrations (Preview)** page, select **Register an Application**.
+1. When the **Register an application page** appears, enter your application's registration information:
+   - In the **Name** section, enter a meaningful application name that will be displayed to users of the app, for example `ConsoleApp-DeltaQuery-DotNet`.
+   - In the **Supported account types** section, select **Accounts in any organizational directory and personal Microsoft accounts (e.g. Skype, Xbox, Outlook.com)**.
+   - Select **Register** to create the application.
+1. On the app **Overview** page, find the **Application (client) ID** value and record it for later. You'll need it to configure the Visual Studio configuration file for this project.
+1. In the list of pages for the app, select **Manifest**, and:
+   - In the manifest editor, set the ``allowPublicClient`` property to **true**
+   - Select **Save** in the bar above the manifest editor.
+1. In the list of pages for the app, select **Authentication**
+   - In the *Suggested Redirect URIs for public clients(mobile,desktop)*, check all the boxes so that the app can work with the MSAL libs used in the application.
+1. In the list of pages for the app, select **API permissions**
+   - Click the **Add a permission** button and then,
+   - Ensure that the **Microsoft APIs** tab is selected
+   - In the *Commonly used Microsoft APIs* section, click on **Microsoft Graph**
+   - In the **Delegated permissions** section, ensure that the right permissions are checked: **Mail.Read**. Use the search box if necessary.
+   - Select the **Add permissions** button
 
-1. In Solution Explorer, select and edit the **App.config** in the ConsoleApplication project.
+### Step 3:  Configure the sample to use your Azure AD tenant
 
-	a. For the `AppPrincipalId` key, replace `To be filled in` with the application ID of your registered Azure application.
+In the steps below, "ClientID" is the same as "Application ID" or "AppId".
 
-<a name="contributing"></a>
-## Contributing ##
+Open the solution in Visual Studio to configure the projects
+
+#### Configure the client project
+
+1. Open the `ConsoleApplication\appsettings.json` file
+1. Find the line where `ClientId` is set and replace the existing value with the application ID (clientId) of the `ConsoleApp-DeltaQuery-DotNet` application copied from the Azure portal.
+1. [optionally] Find the line where `TenantId` is set and replace the existing value with your tenant ID.
```

Comment:
given that you have set the application to support Work and School accounts and Personal accounts, you can leave TenantId to "common".